src/lexitools/commands/correspondences.py

```diff
@@ -1,161 +1,373 @@
+        """
         # see https://github.com/chrzyki/snippets/blob/main/lexibank/install_datasets.py
-        egg_pattern = "git+https://github.com/lexibank/{name}.git#egg=lexibank_{name}"
-        for name in dataset_list:
+        egg_pattern = "git+https://github.com/{org}/{name}.git#egg=lexibank_{name}"
+        needed_install = False
+        for org, name in dataset_list:
             if find_spec("lexibank_" + name) is None:
-                subprocess.run(
-                    ["pip", "install", "-e", egg_pattern.format(name=name)])
+                args = [sys.executable, "-m", "pip", "install",
+                        "-e", egg_pattern.format(org=org, name=name)]
+                subprocess.run(args)
+                needed_install = True
+
+        if needed_install:
+            raise EnvironmentError("Some datasets were not installed. ",
+                                   "I have tried to install them,",
+                                   "please re-run the command now.")
+
         self.datasets = {}
-        for name in dataset_list:
+        for org, name in dataset_list:
             try:
                 self.datasets[name] = get_dataset(name,
                                                   ep="lexibank.dataset").cldf_reader()
             except AttributeError:
                 print("Failed to load", name)
+        if not self.datasets:
+            raise Exception("No dataset loaded")
 
     def iter_table(self, table_name):
+        """ Iter on the rows of a specific table, across all loaded datasets.
+
+        Args:
+            table_name (str): name of a CLDF Wordlist table.
+
+        Yields:
+            rows from all datasets, with an additional "dataset" field indicating the name
+            of the dataset from which the row originates.
+        """
         for name, ds in self.datasets.items():
             for row in ds[table_name]:
                 row["dataset"] = name
                 yield row
 
     def get_table(self, table_name):
+        """ Return all the rows of a specific tablee, across all loaded datasets.
+
+        Args:
+            table_name (str): name of a CLDF Wordlist table.
+
+        Returns:
+            list of rows
+        """
         return list(self.iter_table(table_name))
 
 
-class SoundCorrespsByGenera(object):
+@dataclass(eq=True, frozen=True)
+class Lang:
+    """ A Language has a name, a glottocode, a genus and a family."""
+    genus: str
+    glottocode: str
+    family: str
+    name: str
+
+@dataclass(eq=True, frozen=True)
+class Sound:
+    """ Sound in a correspondence, in a specific genus, language, and context.
+
+    Attributes:
+        sound (str) : the sound which was observed in a correspondence
+        lang (Lang) : the language in which this sound was observed
+        context (str) : the context in which this sound was observed
+    """
+    sound: str
+    lang: Lang
+    context: str
+
+@dataclass
+class Row:
```

Comment:
at the risk of bikeshedding, could this just be 'Word' then?

Reply:
Sure, can do